### ISSUE TITLE
[xray] Terminated actors with queued tasks crashes raylet

### ISSUE THREAD

**Issue by stephanie-wang** [ray-project/ray#2041]
### Describe the problem
Raylet expects actor tasks to be scheduled in order of submission. This is enforced with the actor task counter field in the task spec. However, the `__ray_terminate__` task, submitted when the original actor handle goes out of scope, is scheduled to execute as soon as possible. This means that the terminate task will get scheduled before any other tasks that are queued for the same actor. Raylet then crashes because the terminate task has a higher task counter than expected.

### Source code / logs
To reproduce:
```python
import ray
import time
ray.init(use_raylet=True)

@ray.remote(num_cpus=0)
class Actor(object):
     def __init__(self):
         return
     def foo(self):
         time.sleep(1)
         return

a = Actor.remote()
# Submit a bunch of tasks.
for _ in range(1000):
  a.foo.remote()
# Delete the actor, which triggers the `__ray_terminate__` task.
del a
```

**Comment by pcmoritz** [comment]
As we discussed, one possible fix is to submit the terminate task from a different handle with counter = 0/1 (although this is a little hacky).

**Comment by stephanie-wang** [comment]
#2052 changed the actor Python code so that the `__ray_terminate__` task gets submitted like any other task, so it executes after other methods submitted on the same handle. Therefore, this bug no longer exists.

**Comment by robertnishihara** [comment]
Oh @stephanie-wang you're right, though that was actually unintentional.

As you can see I changed `self._actor_method_call("__ray_terminate__")` to `self.__ray_terminate__.remote()` in 

https://github.com/ray-project/ray/pull/2052/files#diff-26de812097de95458ee3603979d523e0R839

but left the above comment in https://github.com/ray-project/ray/pull/2052/files#diff-26de812097de95458ee3603979d523e0R837.

Could this have caused any behavior changes for RLlib? Since @ericl and @richardliaw had mentioned preferring the immediate termination behavior.
